src/rating/rating.ts

```diff
@@ -218,4 +221,11 @@ export class NgbRating implements ControlValueAccessor,
     this.contexts.forEach(
         (context, index) => context.fill = Math.round(getValueInRange(nextValue - index, 1, 0) * 100));
   }
+
+  private _updateMax() {
+    if (this.max > 0) {
+      this.contexts = Array.from({length: this.max}, (v, k) => ({fill: 0, index: k}));
```

Comment:
This line is now in 2 places (here and in `ngOnInit` - maybe let's pull it into a utility function or a private method?

Reply:
@pkozlowski-opensource - good point, I've moved it into a private method named `_setupContexts()`.